### ISSUE TITLE
no audio/sleep after lid close with kernel 5.3, but fine with lts 4.19

### ISSUE THREAD

**Issue by stefanwiegmann** [raphael/linux-samus#220]
on Archlinux here. I have no audio when booting into 5.3. All works when in 4.19. I did change the order of soundcards in /etc/modprobe.d/audio.conf. The scripts provided here to enable audio, speaker and headphones still work. I just realized, as I did a fresh install. I am not sure if I was in lts exclusively before, so might be not 5.3 specific. Did mainline kernel support change at some point?

**Comment by christianbundy** [comment]
> Did mainline kernel support change at some point?

Not that I'm aware of -- maybe you could bisect? I haven't done a fresh install in a while, but it'd probably be smart to do.

**Comment by stefanwiegmann** [comment]
Haha, feel a little stupid now. Was only missing pulseaudio-alsa and intel-media-driver and then your scripts worked. Guess I got confused why lts was working and the other one didn't. Still doesn't make sense, but it works now. Have one other issue, though. Just changed the title.

If I close my lid it goes to sleep in lts, but shuts down in 5.3.10, seems very odd.

**Comment by stefanwiegmann** [comment]
Okay, fixed that as well. Installed acpi tpm2-tools and enabled module. Closing this. Thanks for reading!

**Comment by drivenbyentropy** [comment]
>  Installed acpi tpm2-tools and enabled module.

@christianbundy, this sounds like it could be related to the sleep issue described in #217. @stefanwiegmann, could you elaborate on the acpi and tmp2-tool part, especially if/where you found the corresponding documentation for this?

Thanks!

**Comment by stefanwiegmann** [comment]
sure, sorry for the short post before :-) I eventually ended up [here](https://bbs.archlinux.org/viewtopic.php?id=176748) There was a similar issue before. I just installed acpi and tmp2-tool and made sure I added `options tpm_tis force=1 interrupts=0` to something like /etc/modprobe.d/suspend.conf

Same for /etc/modprobe.d/audio.conf:
`options snd_soc_sst_bdw_rt5677_mach index=0
options snd-hda-intel index=1`

I guess 5.3 just doesn't autoload (or not with the same options?) the modules I need. I didn't debug what modules were loaded at all or how. I found this and it worked. But if you would like me to, I can go back remove them and see what looks different in dmesg for both kernels. Or whatever else you want me to look at. You should be able to reproduce it by having both kernels installed, default and lts.

I am not sure, but I guess I was on lts for a long time before reinstalling. Might be as well something changed way before 5.3; maybe 4 -> 5?

**Comment by drivenbyentropy** [comment]
@stefanwiegmann Thank you very much for this, I can confirm that installing the `acpi` and `tmp2-tool` packages and adding `options tpm_tis force=1 interrupts=0` to `/etc/modprobe/suspend.d` does indeed solve the reboot issue on the latest kernel (`5.3.11-arch1-1`)!

I will check the sound configuration as well in a bit. 

@christianbundy, since there seem to be a number of small tweaks that need to be done when running the latest kernels, and this repository is (at least in my opinion) still the best place to collect them, I would like to suggest the following to help the remaining Pixel 2 + Linux users out there. We could compile a list of required packages/configurations specific to the 5.x kernel series as an additional page in the wiki and link to it in the `README.MD`. This way, everything required to get full potential of the Pixel would be in one place without needing to search through the closed issues. I would be more than happy to contribute.

Let me know :)

**Comment by christianbundy** [comment]
Please feel free to improve the wiki. :+1:

**Comment by stefanwiegmann** [comment]
@christianbundy , @drivenbyentropy Are you both on arch as well? Maybe we should update the wiki there as it has a dedicated page for the pixel 2. let us know if audio works for you now, please

**Comment by christianbundy** [comment]
That would be great. I'm much more active on that wiki anyway.

**Comment by drivenbyentropy** [comment]
That beats my idea of putting this in the project's wiki 👍. We could still link it to the readme here afterwards.

@stefanwiegmann Yes, I am on arch too. Sound works for me. I add the `enable-audio.sh` , `enable-mic.sh`, and `enable-speakers.sh` scripts to my autostart configuration in Plasma which works for now, but a more persistent solution would be great (im sure im missing something). 

In addition, are you using the `keyboard.sh` script? When I manually execute it, I get the desired functionality but for some reason it gets reset when the pixel goes to sleeps or the plasma screen locks.

**Comment by stefanwiegmann** [comment]
@drivenbyentropy i just picked the chromebook layout in system settings (plasma as well here), not sure what else your desire is missing :-)  

I put the audio commands on custom shortcuts (ctrl+volumeUp,  ctrl+volumeDown, ctrl+mute) and I am fine with pressing them once every reboot. I usually reboot maybe once a month or even less, that's why suspend has to work for me ;-)

**Comment by drivenbyentropy** [comment]
Thanks for the suggestion regarding the keyboard! I must be blind though because I cannot find the Chromebook layout in the keyboard settings. Could you outline the exact steps you did to set this keyboard layout? Thanks again and sorry for the repeated questions.

**Comment by drivenbyentropy** [comment]
Thanks for the suggestion regarding the keyboard! I must be blind though because I cannot find the Chromebook layout in the keyboard settings. Could you outline the  steps you did to set this keyboard layout? Thanks again and sorry for the repeated questions.

**Comment by stefanwiegmann** [comment]
![image](https://user-images.githubusercontent.com/11761918/69022745-4ff7d400-098a-11ea-9ee4-0daa258dcb38.png)
System Settings | Input Devices | Keyboard | Keyboard Model

At the very beginning this was not an option and you needed an aur package. There are so many things becoming "normal" or integrated you often don't realize until reinstalling. Which was my confusion with suspend earlier.

**Comment by drivenbyentropy** [comment]
Got it, thank you! Now I only need to find a way to set up the search key as a modifier between the F1-F10 and the function keys and it would be perfect. This is the default behavior when using the `keyboard.sh` script.

**Comment by stefanwiegmann** [comment]
@christianbundy @raphael @drivenbyentropy I added 2 paragraphs to the arch wiki [here](https://wiki.archlinux.org/index.php/Chromebook_Pixel_2) Please check if that makes sense. I added suspend issue and audio scripts. Maybe the keyboard model / function key - thing would be nice to add there as well?

**Comment by drivenbyentropy** [comment]
@stefanwiegmann Thank you for adding the paragraphs to the wiki, they are great. I will check the keyboard issue in more detail either tomorrow or next weekend and see if I can add these to it as well.

**Comment by drivenbyentropy** [comment]
@stefanwiegmann and @christianbundy, are you by *any* chance still rocking the Pixel Chromebook 2 to this day? If so, I am reaching out because this issue has resurfaced on mine and the old solution of installing `acpi` and `tpm2-tools`, as well as adding options `tpm_tis force=1 interrupts=0` to `/etc/modprobe/suspend.d` has no effect any longer. 

Closing the lid with kernel `5.10.16-arch1-1` will result in the Pixel Chromebook 2 waking up again. Have you by any chance experienced this lately?

I am sorry for reiterating on this old issue but I thought it might be the most appropriate place.

**Comment by stefanwiegmann** [comment]
Hi @drivenbyentropy, @christianbundy, yes, still rocking it here. Although not much traveling going on, right now. I upgraded today to check (last update was 10 days ago). I don't have that issue. tpm2-tools is still installed, but I don't have anything in /etc/modprobe/ for that. It's possible I wiped it and installed from scratch, maybe to switch to btrfs/snapshots.

Wasn't there some hickup with ACPI commands as the reason behind this? Does it work without the  `tpm_tis force=1 interrupts=0`?

I kinda gave up on the mic issue, but other than that, the only "issue" I have is a slowly degrading battery, but that's no surprise with age.

**Comment by drivenbyentropy** [comment]
@stefanwiegmann I know this is reply is way overdue (and I apologize for that), but it took me a while to get to the bottom of the issue. I had implemented a couple of custom systemd scripts to manage auto-mounting some of my nfs shares over wifi which were ultimately interfering with the sleep process and waking up the system in a hard to trace manner. A couple of changes in the dependency hierarchy and I was able to fix the issue.

I do appreciate you getting back to me! Here is to another couple of years of using this ludicrous machine powered by open source.

Cheers!
